src/wallet/wallet.cpp

```diff
@@ -1457,6 +1488,36 @@ void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRe
         // https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Wallet-Transaction-Conflict-Tracking
         SyncTransaction(tx, TxStateInactive{});
     }
+
+    const Txid& conflict_txid = tx->GetHash();
+
+    for (const CTxIn& tx_in : tx->vin) {
+        // If the wallet contains no transactions that spend the same prevout, continue
+        if (mapTxSpends.count(tx_in.prevout) == 0) continue;
+
+        std::pair<TxSpends::const_iterator, TxSpends::const_iterator> range;
+        range = mapTxSpends.equal_range(tx_in.prevout);
+
+        // For each wallet transaction spending this prevout...
+        for (TxSpends::const_iterator _it = range.first; _it != range.second; ++_it) {
+
+            auto try_updating_state = [&](CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) {
+                // Remove the previously conflicting transaction from this wtx's entry
+                wtx.mempool_conflicts.erase(conflict_txid);
+
+                if (wtx.mempool_conflicts.empty()) {
+                    // If this wtx has no other mempool conflicts, it is now considered inactive
+                    if (wtx.isMempoolConflicted()) {
+                        wtx.m_state = TxStateInactive{};
+                        return TxUpdate::CHANGED;
+                    }
+                }
+                return TxUpdate::UNCHANGED;
```

Comment:
In commit "wallet: track mempool conflicts" (2c867784b2a719c9e701e19e389c3adfa7d7faae)

This has the same problems as the new transactionAddedToMempool code above. Would suggest avoiding the need for state transitions, and always updating the mempool_conflicts set recursively for BlockConflicted as well as Inactive transactions:

```c++
if (!wtx.mempool_conflicts.erase(conflict_txid)) return TxUpdate::UNCHANGED;
return TxUpdate::CHANGED;
```

Reply:
Done